File: thera_frontend/lib/models/emotion_entry.dart
class EmotionEntry {
  final int id;
  final String mood; // ex: happy, sad, anxious
  final int intensity; // 1 → 5
  final String note;
  final DateTime createdAt;

  EmotionEntry({
    required this.id,
    required this.mood,
    required this.intensity,
    required this.note,
    required this.createdAt,
  });

  factory EmotionEntry.fromJson(Map<String, dynamic> json) {
    return EmotionEntry(
      id: json['id'],
      mood: json['mood'],
      intensity: json['intensity'],
      note: json['note'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {'mood': mood, 'intensity': intensity, 'note': note};
  }
}

File: demo/src/main/java/com/example/demo/entity/EmotionEntry.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Setter
@Getter
@Entity
@Table(name = "emotion_entry")
public class EmotionEntry {

    // Getters and Setters
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String mood; // e.g., "sad", "happy"

    private int intensity; // 1..5

    @Lob
    private String note;

    private LocalDateTime createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    // Default constructor required by JPA
    public EmotionEntry() {
        this.createdAt = LocalDateTime.now();
    }

    // Optional convenience constructor
    public EmotionEntry(User user, String mood, int intensity, String note) {
        this.user = user;
        this.mood = mood;
        this.intensity = intensity;
        this.note = note;
        this.createdAt = LocalDateTime.now();
    }

}
